climada/util/coordinates.py

```diff
@@ -76,6 +78,256 @@
+            be found. Those are assigned with a -1 index
+
+    Returns:
+        array with so many rows as coordinates containing the centroids
+            indexes
+    """
+    # Construct tree from centroids
+    tree = BallTree(np.radians(centroids), metric='haversine')
+    # Select unique exposures coordinates
+    _, idx, inv = np.unique(coordinates, axis=0, return_index=True,
+                            return_inverse=True)
+
+    # query the k closest points of the n_points using dual tree
+    dist, assigned = tree.query(np.radians(coordinates[idx]), k=1,
+                                return_distance=True, dualtree=True,
+                                breadth_first=False)
+
+    # Raise a warning if the minimum distance is greater than the
+    # threshold and set an unvalid index -1
+    num_warn = np.sum(dist * EARTH_RADIUS_KM > threshold)
+    if num_warn:
+        LOGGER.warning('Distance to closest centroid is greater than %s'
+                       'km for %s coordinates.', threshold, num_warn)
+        assigned[dist * EARTH_RADIUS_KM > threshold] = -1
+
+    # Copy result to all exposures and return value
+    return np.squeeze(assigned[inv])
+
+def interpolate_lines(gdf_lines, point_dist=5):
+    """ Convert a GeoDataframe with LineString geometries to 
+    Point geometries, where Points are placed at a specified distance along the
+    original LineString 
+    
+    Parameters
+    ----------
+    gdf_lines (gpd.GeoDataframe or str) : Geodataframe or filepath from which
+        to read the GeoDataframe
+    point_dist (float) : Distance in metres apart from which the generated  
+        Points should be placed.
+    
+    Returns
+    -------
+    gdf_points with individual Point per row, retaining all other column infos
+        belonging to its corresponding line (incl. line length of original geom.
+        and multi-index referring to original indexing)
+        
+    See also
+    --------
+    * util.coordinates.dist_great_circle_allgeoms()
+    * util.lines_polys_handler.point_exposure_from_lines()
+    """
+
+    if not isinstance(gdf_lines, gpd.GeoDataFrame):
+        gdf_lines = gpd.read_file(gdf_lines)
+        
+    gdf_points = gdf_lines.copy()
+    gdf_points['length'] = np.ones(len(gdf_points))*point_dist
+    gdf_points['length_full'] = dist_great_circle_allgeoms(gdf_points)
+
+    # split line lengths into relative fractions acc to point_dist (e.g. 0, 0.5, 1)
+    gdf_points['distance_vector'] = gdf_points.apply(
+        lambda row: np.linspace(0, 1, num=int(np.ceil(row.length_full/
+                                                      row.length)+1)), axis=1)
+
+    # create MultiPoints along the line for every position in distance_vector
+    gdf_points['geometry'] = gdf_points.apply(
+        lambda row: MultiPoint(
+            [row.geometry.interpolate(dist, normalized=True) 
+             for dist in row.distance_vector]), axis=1)
```

Comment:
The LineString method `interpolate` does interpolation in a Cartesian coordinate system. I think there's not a lot we can do about that, but maybe we can warn the user in the docstring that the result will be inaccurate for LineStrings whose defining points are spaced very far from each other (like end points in Zürich and NYC).

Reply:
Yes, indeed. I added this to the docstring